### PR TITLE
[DOCS] Improve docs for GROUP Share Level

### PR DESCRIPTION
**Pull request by futureltl**
# :mag: Description
## Issue References 🔗

This pull request fixes #3897

## Describe Your Solution 🔧

enrich the description for GROUP Share Level.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: docs/deployment/engine_share_level.md [hunk 1]
```diff
@@ -108,6 +108,57 @@ It follows the [Hadoop GroupsMapping](https://hadoop.apache.org/docs/current/had
 
 The mechanisms of `SparkContext`, `SparkSession` and TTL works similarly to USER share level.
 
+You can use the following steps to configure Hadoop to use LDAP-based group mapping.
+1. Add the properties shown in the example below to the core-site.xml file. You will need to provide the value for the bind user, the bind password, and other properties specific to you LDAP instance, and make sure that object class, user, and group filters match the values specified in your LDAP instance.
```
Comment (pan3793) [inline docs/deployment/engine_share_level.md:112]:
```suggestion
1. Add the properties shown in the example below to the `core-site.xml` file. You will need to provide the value for the bind user, the bind password, and other properties specific to your LDAP instance, and make sure that object class, user, and group filters match the values specified in your LDAP instance.
```

File: docs/deployment/engine_share_level.md [hunk 2]
```diff
@@ -108,6 +108,57 @@ It follows the [Hadoop GroupsMapping](https://hadoop.apache.org/docs/current/had
 
 The mechanisms of `SparkContext`, `SparkSession` and TTL works similarly to USER share level.
 
+You can use the following steps to configure Hadoop to use LDAP-based group mapping.
```
Comment (pan3793) [inline docs/deployment/engine_share_level.md:111]:
```suggestion
Here is an example to configure `HadoopGroupProvider` to use LDAP-based group mapping.
```

File: docs/deployment/engine_share_level.md [hunk 3]
```diff
@@ -108,6 +108,57 @@ It follows the [Hadoop GroupsMapping](https://hadoop.apache.org/docs/current/had
...
+  <value>memberuid</value>
+</property>
+
+<property>
+  <name>hadoop.security.group.mapping.ldap.search.attr.group.name</name>
+  <value>cn</value>
+</property>
+```
+2. Use the applicable instructions to re-start the HDFS NameNode and the YARN ResourceManager.
+3. Verify LDAP group mapping by running the **hdfs groups** command. This command will fetch groups from LDAP for the current user. Note that with LDAP group mapping configured, the HDFS permissions can leverage groups defined in LDAP for access control.
```
Comment (pan3793) [inline docs/deployment/engine_share_level.md:160]:
```suggestion
3. Verify LDAP group mapping by running the `hdfs groups` command. This command will fetch groups from LDAP for the current user. Note that with LDAP group mapping configured, the HDFS permissions can leverage groups defined in LDAP for access control.
```

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
@futureltl thanks for supplying an example for configuring Hadoop Group Mapping, the style issue could be auto-fixed by `dev/reformat`. As you touched the docs, could you update it to reflect https://github.com/apache/kyuubi/pull/3897 too?

**Comment by futureltl** [comment]
> @futureltl thanks for supplying an example for configuring Hadoop Group Mapping, the style issue could be auto-fixed by `dev/reformat`. As you touched the docs, could you update it to reflect #3897 too?

**Comment by pan3793** [comment]
@futureltl do you forget to push your change?

File: docs/deployment/engine_share_level.md [hunk 4]
```diff
@@ -108,6 +108,59 @@ It follows the [Hadoop GroupsMapping](https://hadoop.apache.org/docs/current/had
 
 The mechanisms of `SparkContext`, `SparkSession` and TTL works similarly to USER share level.
 
+Here is an example to configure `HadoopGroupProvider` to use LDAP-based group mapping.
+1. Add the properties shown in the example below to the `core-site.xml` file. You will need to provide the value for the bind user, the bind password, and other properties specific to your LDAP instance, and make sure that object class, user, and group filters match the values specified in your LDAP instance.
+
+```
```
Comment (bowenliang123) [inline docs/deployment/engine_share_level.md:114]:
```suggestion
```xml
```
Adding language hint for better UI style by default.

**Review by bowenliang123 — APPROVED** [review]

File: docs/deployment/engine_share_level.md (hunk 4 above)
Comment (futureltl) [reply docs/deployment/engine_share_level.md]: done

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**